clang/test/SemaSYCL/sycl-cconv.cpp

```diff
@@ -1,14 +1,20 @@
 // RUN: %clang_cc1 -triple spir64-unknown-windows-sycldevice -aux-triple x86_64-pc-windows-msvc -fsycl-is-device -fsyntax-only -verify %s
 
-//expected-warning@+1 {{'__cdecl' calling convention is not supported for this target}}
+// expected-no-warning@+1
 __inline __cdecl int printf(char const* const _Format, ...) { return 0; }
+// expected-no-warning@+1
+__inline __cdecl __attribute__((sycl_device)) int foo() { return 0; }
+// expected-no-warning@+1
+__inline __cdecl __attribute__((sycl_kernel)) int moo() { return 0; }
```

Comment:
`sycl_kernel` is supposed to be applied only to function templates.
There is a number of restrictions on using this attribute, which I'm going to enforce with this patch: https://reviews.llvm.org/D60455.

I suggest adding `__cdecl` calling convention to `kernel_single_task` and remove `sycl_kernel` attribute from the definition of `moo`. Also I would call `moo` from the SYCL kernel. There is no example of the "successful" call of a `__cdel` function. We test only `printf` function call, which is not allowed.

Reply:
Done, thanks!